web/src/lib/domain-logic/essentialQuestions.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  emptyIndustrySpecificData,
+  Industry,
+  IndustrySpecificData,
+  LookupIndustryById,
+  ProfileData,
+} from "@businessnjgovnavigator/shared";
+
+type EssentialQuestionFunction = (industry: Industry) => boolean;
+interface EssentialQuestion
+  extends Omit<EssentialQuestionsProps<ProfileData[keyof ProfileData]>, "fieldName" | "choices"> {
+  fieldName: keyof IndustrySpecificData;
+  validationFunc: EssentialQuestionFunction;
+  reset: boolean;
+  convenienceFunc: (industryId: string | undefined) => boolean;
+}
+
+export const getResetIndustrySpecificData = (industryId: string | undefined) => {
+  const industry = LookupIndustryById(industryId);
+  return EssentialQuestions.filter((eQ) => {
+    return eQ.reset && !eQ.validationFunc(industry);
+  }).reduce((reducer, eQ) => {
+    return { ...reducer, [eQ.fieldName]: emptyIndustrySpecificData[eQ.fieldName] };
+  }, {} as Partial<IndustrySpecificData>);
+};
+
+export const getValidationFuncByFieldName = (fieldName: ProfileLabelField) => {
+  return EssentialQuestions.find((eQ) => {
+    eQ.contentFieldName == fieldName || eQ.fieldName == fieldName;
+  })?.validationFunc;
+};
+
+export const validateByIndustryIdAndFieldName = (fieldName: ProfileLabelField) => {
+  return (industryId: string | undefined): boolean => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return EssentialQuestions.find((eQ) => {
+      return eQ.fieldName == fieldName || eQ.contentFieldName == fieldName;
+    })!.validationFunc(LookupIndustryById(industryId));
+  };
+};
+
+export const hasEssentialQuestion = (industryId: string | undefined): boolean => {
+  const industry = LookupIndustryById(industryId);
+  const essentialQuestionResults = EssentialQuestions.map((essentialQuestionFunction) => {
+    return essentialQuestionFunction.validationFunc(industry);
+  });
+  return essentialQuestionResults.includes(true);
+};
+
+export const isInterstateLogisticsApplicable = validateByIndustryIdAndFieldName("interstateLogistics");
+export const isInterstateMovingApplicable = validateByIndustryIdAndFieldName("interstateMoving");
+export const isCarServiceApplicable = validateByIndustryIdAndFieldName("carService");
+export const isCertifiedInteriorDesignerApplicable =
+  validateByIndustryIdAndFieldName("certifiedInteriorDesigner");
+export const isChildcareForSixOrMoreApplicable = validateByIndustryIdAndFieldName("isChildcareForSixOrMore");
+export const isCpaRequiredApplicable = validateByIndustryIdAndFieldName("requiresCpa");
+export const isCannabisLicenseApplicable = validateByIndustryIdAndFieldName("cannabisLicenseType");
+export const isLiquorLicenseApplicable = validateByIndustryIdAndFieldName("liquorLicense");
+export const isProvidesStaffingServicesApplicable =
+  validateByIndustryIdAndFieldName("providesStaffingService");
+export const isRealEstateAppraisalManagementApplicable = validateByIndustryIdAndFieldName(
+  "realEstateAppraisalManagement"
```

Comment:
going through these, it seems like `validationFunc` and `convenienceFunc` are the same? Why two fields?